ui/packages/consul-ui/tests/integration/components/hashicorp-consul-test.js

```diff
@@ -21,4 +22,70 @@ module('Integration | Component | hashicorp consul', function(hooks) {
 
     assert.dom('*').hasText('template block text');
   });
+
+  module('datacenter dropdown', function(hooks) {
+    hooks.beforeEach(function() {
+      const router = this.owner.lookup('router:main');
+      // we need to patch router as it relies on our custom location
+      // implementation which isn't initialized correctly in rendering tests
+      router.reopen({
+        get location() {
+          return {
+            hrefTo() {},
+          };
+        },
+        get currentRouteName() {
+          return 'testing';
+        },
+      });
+
+      // patch router service as there is no active route and helpers we use
+      // rely on this method.
+      const routerService = this.owner.lookup('service:router');
+      routerService.reopen({
+        isActive() {
+          return false;
+        },
+      });
+    });
+
+    test('it does not display a dropdown when only one dc is available', async function(assert) {
+      const dcs = [
+        {
+          Name: 'dc-1',
+        },
+      ];
+      this.set('dcs', dcs);
+      this.set('dc', dcs[0]);
+
+      await render(hbs`<HashicorpConsul @dcs={{this.dcs}} @dc={{this.dc}} />`);
+
+      assert
+        .dom('[data-test-datacenter-menu]')
+        .doesNotExist('datacenter dropdown is not displayed in nav');
+
+      assert.dom('[data-test-datacenter]').hasText('dc-1', 'Datecenter name is displayed in nav');
+    });
+
+    test('it does displays a dropdown when more than one dc is available', async function(assert) {
+      const dcs = [
+        {
+          Name: 'dc-1',
+        },
+        {
+          Name: 'dc-2',
+        },
+      ];
+      this.set('dcs', dcs);
+      this.set('dc', dcs[0]);
+
+      await render(hbs`<HashicorpConsul @dcs={{this.dcs}} @dc={{this.dc}} />`);
+
+      assert
+        .dom('[data-test-datacenter]')
+        .doesNotExist('we are displaying more than just the name of the first dc');
+
+      assert.dom('[data-test-datacenter-menu]').exists('datacenter dropdown is displayed');
+    });
+  });
```

Comment:
Am I right in thinking these tests are testing `if arr.length > 1`?

Wondered if there was anything else we are testing here?

I think if we want to test this, we should move these tests to the datacenter::selector (if we implement the suggestions from above)

Reply:
Ah ok, so my main point here is what is the logic we are testing, it looks like we are testing `arr.length > 1`?

FYI: Our testing approach closely aligns with BoundaryUIs:

> We strive for "the right amount of testing". Use test coverage as a guide to help you identify untested high-value code.

This isn't a huge concern, but just wanted to double check the logic we are testing here and I'm wondering whether this fits with the "right amount of testing" idea, what are your thoughts on that?

I'm actually more interested in why we need to patch the router in a component test? Is it because of href-to?

Reply:
This is a test that makes sure that when we have one datacenter we display the name only when we have multiple we display the menu - which is the change that I'm making in this PR. I'm not adding a test for the message because that's just the template layer of Ember that I would be testing.

The patch was necessary because the helpers and how routing is setup breaks when testing the `HashicorpConsul`-component in isolation. With the latest addition I moved to a component test of the selector component itself because now everything is contained in the selection component. No more router patching necessary.

Reply:
> I'm not adding a test for the message because that's just the template layer of Ember that I would be testing.

Awesome, yes that sounds great!

> The patch was necessary because the helpers

Gotcha, thats what I was wondering, if we needed to patch here, I would have preferred to patch the helper. 

Glad now we've moved this around a little we no longer need to patch the router, thanks for doing that!

Yet to look at this altogether now, but this sounds much better 👏